lib/tessel/deploy.js

```diff
@@ -124,58 +114,70 @@ Tessel.prototype.memoryInfo = function() {
       .catch(reject);
   });
 };
-/*
-  Run a script on a Tessel
-    param opts: unused so far
-*/
-Tessel.prototype.deployScript = function(opts) {
+/**
+ * Deploy project to a Tessel 2.
+ * Language: *
+ *
+ * @return {Promise}
+ */
+Tessel.prototype.deploy = function(opts) {
   // Only an _explicit_ `true` will set push mode
   var isPush = opts.push === true;
   var entryPoint = opts.entryPoint;
 
+  // Resolve the application's language/runtime
+  opts.lang = deployment.resolveLanguage(opts.lang || opts.entryPoint);
```

Comment:
This operation makes a best possible guess at what language your application is written in. [deployment.resolveLanguage](https://github.com/rwaldron/t2-cli/blob/deployment-reorg/lib/tessel/deployment/index.js)